kornia/utils/draw.py

```diff
@@ -0,0 +1,53 @@
+from typing import Optional, Union
+
+import torch
+
+
+def rectangle(
+        image: torch.Tensor,
+        boxes: torch.Tensor,
+        color: Optional[Union[float, torch.Tensor]] = 1.0,
+        fill: Optional[bool] = False) -> torch.Tensor:
+    """Draws N rectangles on a batch of image tensors
+
+    Args:
+        image (torch.Tensor): is tensor of BxCxHxW
+        boxes (torch.Tensor): represents number of rectangles to draw in BxNx4
+            N is the number of boxes to draw per batch
+            4 is in (top_left.y, top_left.x, bot_right.y, bot_right.x) ie h1, w1, h2, w2
+        color (int or torch.Tensor): can be a float or tensor if image channels > 1
+            if float is used and channels > 1, float will be broadcasted to all channels
+        fill: bool is a flag used to fill the boxes with color if True
+
+    Returns:
+        torch.Tensor: This operation modifies image inplace but also returns the drawn tensor for convenience
+    """
+
+    batch, c, h, w = image.shape
+    _, num_boxes, num_points = boxes.shape
+
+    # clone boxes, in case it's been expanded assignment from clipping causes problems
+    boxes = boxes.clone()
+
+    # clip boxes to hxw bounds
+    boxes[:, :, ::2] = torch.clamp(boxes[:, :, ::2], 0, h - 1)
+    boxes[:, :, 1::2] = torch.clamp(boxes[:, :, 1::2], 0, w - 1)
+    diff = boxes[:, :, 2:] - boxes[:, :, :2]
+    if torch.any(diff < 0):
```

Comment:
wouldn't be this very expensive to compute every time ?

Reply:
@edgarriba I don't believe torch.any should be very expensive, but maybe you're referring to the strided boxes on line 33 and 34?

Reply:
I refer to the `torch.any` - We also had a short discussion about this kind of value checks with @shijianjian and @FGeri for the augmentations e.g. to check whether a tensor image is bounded between 0/1 but since we are dealing with batched tensors, it can be a bit tricky to determine which sample within the batch is the affected by that condition. If it's only one it's fine, but in case of having multiple not sure what's the proper approach to follow. This also goes to the direction of tensor oriented programming practices.